lib/ansible/plugins/inventory/linode.py

```diff
@@ -125,18 +132,61 @@ def _filter_by_config(self, regions, types):
                 if instance.type.id in types
             ]
 
-    def _add_instances_to_groups(self):
+    def _get_instance_ip(self, instance, private=False):
+        """Returns the public or private IP address of the instance"""
+        if instance.ipv4 and len(instance.ipv4) > 0:
+            result_ip = None
+            for ip in instance.ipv4:
+                if not ipaddress.ip_address(ip).is_private and not private:
+                    result_ip = ip
+                elif ipaddress.ip_address(ip).is_private and private:
+                    result_ip = ip
+            if result_ip:
+                return result_ip
+        else:
+            raise AnsibleError(
+                'Instance IPv4 address is missing or empty'
+            )
+
+    def _add_instances_to_groups(self, instance_access):
         """Add instance names to their dynamic inventory groups."""
         for instance in self.instances:
-            self.inventory.add_host(instance.label, group=instance.group)
+            if instance_access == 'public_ip':
+                host_identifier = self._get_instance_ip(instance, private=False)
+            elif instance_access == 'private_ip':
+                host_identifier = self._get_instance_ip(instance, private=True)
+            elif instance_access == 'hostname':
+                host_identifier = instance.label
+            else:
```

Comment:
Same as https://github.com/ansible/ansible/pull/51196#discussion_r266537178, we can drop it?

Reply:
Drop what? Even using "choice" we have to set the `host_identifier` based on the choice?

Reply:
Ah, right, got it :+1: